test/gen_rmq_publisher_test.exs

```diff
@@ -149,6 +150,159 @@ defmodule GenRMQ.PublisherTest do
     end
   end
 
+  describe "consumer_count/2" do
+    setup do
+      with_test_publisher(Default)
+    end
+
+    test "should return the correct number of consumers when a valid queue is provided", %{publisher: publisher_pid} do
+      assert 0 == GenRMQ.Publisher.consumer_count(publisher_pid, @out_queue)
+    end
+
+    test "should raise an error when an invalid queue is provided", %{publisher: publisher_pid} do
+      Process.flag(:trap_exit, true)
+
+      catch_exit do
+        GenRMQ.Publisher.consumer_count(publisher_pid, @invalid_queue)
+      end
+
+      assert_receive {:EXIT, ^publisher_pid, _error_details}
+    end
+  end
+
+  describe "empty?/2" do
+    setup do
+      with_test_publisher(Default)
+    end
+
+    test "should return true if the provided queue is empty", %{publisher: publisher_pid} do
+      assert true == GenRMQ.Publisher.empty?(publisher_pid, @out_queue)
+    end
+
+    test "should return false if the provided queue is not empty", %{publisher: publisher_pid} = context do
+      messages = generate_messages(5)
+
+      Enum.each(messages, fn message ->
+        :ok = GenRMQ.Publisher.publish(publisher_pid, Jason.encode!(message))
+      end)
+
+      Assert.repeatedly(fn -> assert out_queue_count(context) == 5 end)
+      assert false == GenRMQ.Publisher.empty?(publisher_pid, @out_queue)
+    end
+
+    test "should raise an error if the provided queue is invalid", %{publisher: publisher_pid} do
+      Process.flag(:trap_exit, true)
+
+      catch_exit do
+        GenRMQ.Publisher.empty?(publisher_pid, @invalid_queue)
+      end
+
+      assert_receive {:EXIT, ^publisher_pid, _error_details}
+    end
+  end
+
+  describe "message_count/2" do
+    setup do
+      with_test_publisher(Default)
+    end
+
+    test "should return the correct number of messages when a valid empty queue is provided", %{
+      publisher: publisher_pid
+    } do
+      assert 0 == GenRMQ.Publisher.message_count(publisher_pid, @out_queue)
+    end
+
+    test "should return the correct number of messages when a valid queue is provided",
+         %{publisher: publisher_pid} = context do
+      messages = generate_messages(5)
+
+      Enum.each(messages, fn message ->
+        :ok = GenRMQ.Publisher.publish(publisher_pid, Jason.encode!(message))
+      end)
+
+      Assert.repeatedly(fn -> assert out_queue_count(context) == 5 end)
+      assert length(messages) == GenRMQ.Publisher.message_count(publisher_pid, @out_queue)
+    end
+
+    test "should raise an error when an invalid queue is provided", %{publisher: publisher_pid} do
+      Process.flag(:trap_exit, true)
```

Comment:
`with_test_publisher` already does `Process.flag(:trap_exit, true)`, so I guess no need to call it here once again. Do we also need to `catch_exit` while trapping exits?

Reply:
Good call...I missed that! Updated the tests and removed the redundant calls.